### DISCUSSION TITLE
Implement the DXGI Flip Model to reduce input latency

### DISCUSSION THREAD

**Discussion by SIUUUUUUUUUU17** [ppy/osu#29900] [Ideas] [answered]
![image](https://github.com/user-attachments/assets/2dc91881-b644-475f-95cc-e6e433b2f70a)

Description from the Assetto Corsa Content Manager launcher

With this, input latency might be improved when playing windowed or borderless. This will not work on Windows 7 or earlier however

**Comment by peppy** [comment]
I'm pretty sure we either already do this, or we tried this and it didn't work well for some reason (cc @smoogipoo).

**Reply by smoogipoo** [reply, marked as answer]
Dunno how you're seeing this (perhaps using OpenGL?), but yes we do use this in the d3d backend. No the latency is not and can not ever be as low as exclusive fullscreen.

**Comment by Ergonom1337** [comment]
In windows 10+ with HAGS and FSO enabled (which is like that by default) all d3d apps are working in borderless mode, there is no exclusive fullscreen. If you need DXGI flip model for OGL you can use "Vulkan/OpenGL Present Method - Prefer Layered on DXGI Swapchain" in nvidia app graphics settings for game you need (this also works in osu stable and fixes issues with streaming in discord btw).
